### PR TITLE
Fix `is_separator()` on disconnected graphs

### PR DESCRIPTION
**Pull request by szhorvat**
The R (but not C) documentation includes:

> In the special case of a fully connected graph with $n$ vertices, each set of $n−1$ vertices is considered to be a vertex separator.

This is probably for consistency with the usual convention that assigns vertex connectivity $n-1$ to $K_n$.

I am not in favour of maintaining this definition of separators in complete graphs. However, we will need to make very sure that if we change this, we won't break any functions that call `igraph_is_separator()`.

### CODE REVIEW

**Comment by szhorvat** [comment]
@ntamas, can you please have a first read-through? What I am looking for at this stage is if the approach and comments are clear enough. Feel free to suggest more comments.

`igraph_is_minimal_separator()` needs more tests.

File: src/connectivity/separators.c [hunk 1]
```diff
@@ -34,97 +34,253 @@
...
+    IGRAPH_CHECK(igraph_vit_create(graph, S, &vit));
+    IGRAPH_FINALLY(igraph_vit_destroy, &vit);
+
+    IGRAPH_VECTOR_CHAR_INIT_FINALLY(&mark, vcount);
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&neis, 10);
+    IGRAPH_DQUEUE_INT_INIT_FINALLY(&Q, 100);
+
+#define VISITED(x) (VECTOR(mark)[x] & 1) /* Was x visited? */
+#define SET_VISITED(x) (VECTOR(mark)[x] |= 1) /* Mark x as visited. */
+#define INS(x) (VECTOR(mark)[x] & 2) /* Is x in S? */
```
Comment (ntamas) [inline src/connectivity/separators.c:80]: I'd call this `IN_S`; `INS` makes me think about insertion.

**Review by ntamas — COMMENTED** [review]
Looks clear enough to me so far, I've posted one comment related to naming only.

File: src/connectivity/separators.c (hunk 1 above)
Comment (szhorvat) [reply src/connectivity/separators.c:80]: Sure, I'll change that.

The minimal separator test is wrong, I'm working on fixing that now.

**Comment by szhorvat** [comment]
@ntamas Can you see if there's an easy fix for the CodeQL failure, which is unrelated to this PR?

I believe this is now correct, but let me sleep on it once. I'll let you know when it's ready for a check. I meant to write a test that first decomposes the graph into components, and tests each one. But `igraph_decomponse()` doesn't keep track of the vertex IDs between the original graph and components, unfortunately, so this would take an inconvenient amount of work.

Regarding the handling of size $n-1$ vertex subsets of the $K_n$ complete graph: the note in the R docs was actually incorrect. I'm going to remove it.

**Comment by szhorvat** [comment]
This is ready for review now.

**Review by ntamas — APPROVED** [review]
Looks good to me after reading the code. I trust that the tests are extensive enough to cover the corner cases that I wasn't thinking about ;)

**Comment by szhorvat** [comment]
OK, I'll merge this an later I'll do #2512 and #2516 which will make testing easier. #2516 will be an internal function for now. We can add the extra features to the existing public functions in 1.0.
